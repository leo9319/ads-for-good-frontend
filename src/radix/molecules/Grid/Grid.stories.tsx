import React from 'react';
import Grid from './Grid';
import { Meta, StoryObj } from '@storybook/react';
import { Theme } from '@radix-ui/themes';
import MediaCardTemplate from '@components/organisms/mediaCardTemplate';

const meta = {
  title: 'Library/Molecules/Grid',
  component: Grid,
  decorators: [
    Story => (
      <Theme>
        <Story />
      </Theme>
    ),
  ],
  parameters: {
    layout: 'fullscreen',
  },
  tags: ['autodocs'],
  argTypes: {
    columns: {
      control: 'select',
      options: ['1', '2', '3', '4', '5', '6', '7', '8', '9'],
    },
    gap: {
      control: 'select',
      options: ['1', '2', '3', '4', '5', '6', '7', '8', '9'],
    },
    className: {
      control: 'text',
    },
  },
} satisfies Meta<typeof Grid>;

export default meta;

type Story = StoryObj<typeof meta>;

const TextAndImageContent = () => (
  <>
    <MediaCardTemplate
      date="Mar 4, 2025"
      id="sample-123"
      imageSrc="https://images.pexels.com/photos/2113709/pexels-photo-2113709.jpeg"
      imgAspectRatio="1:1"
      isArticle
      mediaType="image"
      onClick={() => {}}
      size="1"
      subHead="Proin congue lorem sit amet augue feugiat, eu placerat odio interdum. Mauris tincidunt arcu ut justo sollicitudin, eget volutpat magna ultrices. Nam tempus dapibus vehicula. Morbi blandit molestie metus vitae rhoncus. In in turpis sed ante euismod congue. Duis eu risus a nisl dapibus ultricies. Sed ultrices auctor lorem a finibus."
      title="Video: Burundi schools serve up success one plate at a time"
    />
    <MediaCardTemplate
      date="Mar 4, 2025"
      id="sample-123"
      imageSrc="https://images.pexels.com/photos/2113709/pexels-photo-2113709.jpeg"
      imgAspectRatio="1:1"
      isArticle
      mediaType="image"
      onClick={() => {}}
      size="1"
      subHead="Proin congue lorem sit amet augue feugiat, eu placerat odio interdum. Mauris tincidunt arcu ut justo sollicitudin, eget volutpat magna ultrices. Nam tempus dapibus vehicula. Morbi blandit molestie metus vitae rhoncus. In in turpis sed ante euismod congue. Duis eu risus a nisl dapibus ultricies. Sed ultrices auctor lorem a finibus."
      title="Video: Burundi schools serve up success one plate at a time"
    />
    <MediaCardTemplate
      date="Mar 4, 2025"
      id="sample-123"
      imageSrc="https://images.pexels.com/photos/2113709/pexels-photo-2113709.jpeg"
      imgAspectRatio="1:1"
      isArticle
      mediaType="image"
      onClick={() => {}}
      size="1"
      subHead="Proin congue lorem sit amet augue feugiat, eu placerat odio interdum. Mauris tincidunt arcu ut justo sollicitudin, eget volutpat magna ultrices. Nam tempus dapibus vehicula. Morbi blandit molestie metus vitae rhoncus. In in turpis sed ante euismod congue. Duis eu risus a nisl dapibus ultricies. Sed ultrices auctor lorem a finibus."
      title="Video: Burundi schools serve up success one plate at a time"
    />
    <MediaCardTemplate
      date="Mar 4, 2025"
      id="sample-123"
      imageSrc="https://images.pexels.com/photos/2113709/pexels-photo-2113709.jpeg"
      imgAspectRatio="1:1"
      isArticle
      mediaType="image"
      onClick={() => {}}
      size="1"
      subHead="Proin congue lorem sit amet augue feugiat, eu placerat odio interdum. Mauris tincidunt arcu ut justo sollicitudin, eget volutpat magna ultrices. Nam tempus dapibus vehicula. Morbi blandit molestie metus vitae rhoncus. In in turpis sed ante euismod congue. Duis eu risus a nisl dapibus ultricies. Sed ultrices auctor lorem a finibus."
      title="Video: Burundi schools serve up success one plate at a time"
    />
  </>
);

export const CardGrid: Story = {
  args: {
    children: (
      <div style={{ padding: '20px 0' }}>
        <Grid
          columns={{ initial: '1', lg: '3', md: '2', sm: '2', xs: '1' }}
          gap={{ initial: '4', lg: '4', md: '3', sm: '4', xs: '4' }}
        >
          <TextAndImageContent />
        </Grid>
      </div>
    ),
  },
} as never;
