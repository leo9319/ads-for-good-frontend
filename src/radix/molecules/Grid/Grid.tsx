import React from 'react';
import { Grid as RadixGrid, GridProps as Props } from '@radix-ui/themes';
import { CommonSizes } from '@internal/types/common/style';
import { classNames, getModuleClasses } from '@utils/common/classNames';
import Flex from '@radix-styles/atoms/Flex/Flex';

import styles from './Grid.module.scss';

/**
 * @interface
 * The Grid props type of {@link Grid}
 *
 * @property {Props} props - The props of the Grid component
 */

export type GridProps = Props & {
  /**
   * The card count for the single row
   * @property
   */
  columns?: CommonSizes;
  /**
   * The gap between the cards
   * @property
   * @defaultValue { initial: '2', xl: '5', lg: '5', md: '4', sm: '2', xs: '2' }
   */
  gap?: CommonSizes;
  /**
   * class name for the grid
   * @property
   */
  className?: string;
};

/**
 *
 * Grid component to display the Cards.
 *
 * @group Molecule
 * @category Radix Component
 *
 * @param {GridProps} props - The Grid props
 * @returns {ReactElement} - Grid Component
 *
 */

export const Grid = ({
  columns,
  gap = { initial: '2', xl: '5', lg: '5', md: '4', sm: '2', xs: '2' },
  className,
  children,
  px = { initial: '4', xl: '9', lg: '9', md: '4', sm: '4', xs: '4' },
  ...props
}: GridProps) => {
  const classes = getModuleClasses(className?.trim(), styles);
  const wrapperClass = classNames(
    styles.gridContainer,
    'gridContainer',
    classes
  );

  return (
    <Flex display="flex" justify="center">
      <RadixGrid
        className={wrapperClass}
        columns={columns}
        gap={gap}
        px={px}
        {...props}
      >
        {children}
      </RadixGrid>
    </Flex>
  );
};

export default Grid;
