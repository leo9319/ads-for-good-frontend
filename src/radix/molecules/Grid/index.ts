export type { GridProps } from './Grid';
export { Grid } from './Grid';
export { default } from './Grid';
