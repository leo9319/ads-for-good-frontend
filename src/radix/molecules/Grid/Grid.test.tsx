import React from 'react';
import Grid from './Grid';
import { render } from '@testing-library/react';

describe('Grid', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<Grid />);
    expect(baseElement).toMatchSnapshot();
  });
});
