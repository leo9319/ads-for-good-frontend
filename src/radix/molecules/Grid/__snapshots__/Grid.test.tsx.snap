// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should render successfully 1`] = `
<body>
  <div>
    <div
      class="rt-Grid rt-r-gtc-3 rt-r-gap-3 gridContainer gridContainer"
    />
  </div>
</body>
`;
